plugins/datastreamer_plugin/fastdds/Participant.cpp

```diff
@@ -167,29 +163,86 @@ void Participant::create_subscription(
     auto topic_type = discovery_database_->find(topic_name);
     if (topic_type == discovery_database_->end())
     {
-        WARNING("Topic " << topic_name << " has not been discovered not exist, so type unknowon");
+        WARNING("Topic " << topic_name << " has not been discovered not exist, so type unknown");
         throw InconsistencyException("Trying to create Data Reader in a non existing topic: " + topic_name);
     }
 
-    // Get TypeName associated with topic name
-    std::string type_name = discovery_database_->operator [](topic_name).first;
+    DataTypeNameType type_name = discovery_database_->operator [](topic_name).first;
+    DynamicType::_ref_type dyn_type;
+
+    // Check if type is already registered
+    if (discovery_database_->operator [](topic_name).second == false)
+    {
+        WARNING("Type " << topic_name << " has not been registered yet");
+        throw InconsistencyException("Trying to create Data Reader in a non registered type: " + topic_name);
+    }
+
+    // Check if type is registered or not in participant. If not, register it
+    if (!participant_->find_type(type_name))
+    {
+        // __FLAG__
+        DEBUG("Type info not registered in participant for topic " << topic_name);
+        //////////////////////////
+
+        // Type information is available but not registered in participant
+        // Types manually loaded (through XML file) are registered in participant when loaded, so this case is not possible
+        // The only case is that the type info has been discovered automatically and the type Id has been saved in dyn_types_info_
+        // Get TypeName and DataTypeId associated with topic name
+        DataTypeId type_id = dyn_types_info_->operator [](topic_name).second;
+
+        // Get Dyn Type and register discovered type
+        xtypes::TypeObject type_object;
+        if (RETCODE_OK != DomainParticipantFactory::get_instance()->type_object_registry().get_type_object(
+                    type_id, type_object))
+        {
+            EPROSIMA_LOG_ERROR(PARTICIPANT, "Error getting type object for type " << type_name);
+            return;
+        }
+        dyn_type = DynamicTypeBuilderFactory::get_instance()->create_type_w_type_object(
+            type_object)->build();
+        TypeSupport dyn_type_support(new DynamicPubSubType(dyn_type));
+        dyn_type_support.register_type(participant_);
+    }
+    else
+    {
+        // TODO (Carlosespicur): Check if it can be done simpler. Can we avoid to create a new dyn type?
```

Comment:
The creation of a new `DynamicType` is indeed necessary. The reason for this is that we need to create `DynamicData` within the `ReaderHandler`, and this `DynamicData` requires a corresponding `DynamicType` to define its structure.

Reply:
Yeah, outdated comment that I forgot to remove. Thanks for the explanation.